plugins/parsers/csv/parser.go

```diff
@@ -208,8 +251,10 @@ outer:
 
 	// will default to plugin name
 	measurementName := p.MetricName
-	if recordFields[p.MeasurementColumn] != nil && recordFields[p.MeasurementColumn] != "" {
-		measurementName = fmt.Sprintf("%v", recordFields[p.MeasurementColumn])
+	if p.MeasurementColumn != "" {
+		if recordFields[p.MeasurementColumn] != nil && recordFields[p.MeasurementColumn] != "" {
```

Comment:
how can you compare this without a type assertion?

Reply:
I wasn't sure about this existing code, so I added some tests, seems to be working.  The conditional I added fixes a bug where the measurement name could be set to the value of an empty string named column.

Found this in the spec:

> Interface values are comparable. Two interface values are equal if they have identical dynamic types and equal dynamic values or if both have value nil.